File: TwoDimension.java
package physics.mechanics.motion.generic;

import physics.mechanics.object.Object;

public class TwoDimension {

	private Object object;
	private double distanceX, accX, distanceY, accY;
	private double forceX, forceY;

	public void initiate(Object o) {
		this.object = o;
	}

	//X dimension
	public double calculateAccelerationX(double v1X, double v2X, double t) {
		accX = (v2X-v1X)/t;
		return accX;
	}

	public double calculateDistanceX(double v1X, double v2X, float t) {
		if(accX==0d)
			calculateAccelerationX(v1X, v2X, t);
		distanceX = v1X*t + (double)1/2*accX*t*t;
		return distanceX;
	}

	public double calculateForceX(double v1X, double v2X, double t) {
		if(accX==0d)
			calculateAccelerationX(v1X, v2X, t);
		forceX = object.getMass()*accX;
		return forceX;
	}

	//Y dimension
	public double calculateAccelerationY(double v1Y, double v2Y, double t) {
		accY = (v2Y-v1Y)/t;
		return accY;
	}

	public double calculateDistanceY(double v1Y, double v2Y, float t) {
		if(accY==0d)
			calculateAccelerationY(v1Y, v2Y, t);
		distanceY = v1Y*t + (double)1/2*accY*t*t;
		return distanceY;
	}

	public double calculateForceY(double v1Y, double v2Y, double t) {
		if(accY==0d)
			calculateAccelerationY(v1Y, v2Y, t);
		forceY = object.getMass()*accY;
		return forceY;
	}

	//final
	public double calculateAcceleration() {
		return Math.sqrt(Math.pow(accX,2)+Math.pow(accY,2));
	}
	
	public double calculateDistance() {
		return Math.sqrt(Math.pow(distanceX,2)+Math.pow(distanceY,2));
	}
	
	public double calculateForce() {
		return Math.sqrt(Math.pow(forceX,2)+Math.pow(forceY, 2));
	}

}

File: TwoDimension.py
import math
import numpy as nm
import matplotlib.pyplot as plt

# along x-axis
def calculateAccelerationX(v2, t):
    accX = (v2 - iVelocityX)/t
    return accX

def calculateDistanceX(v2, t):
    if accX == 0:
        calculateAccelerationX(v2,t)
    distanceX = iVelocityX*t + accX*t*t/2
    return distanceX

def calculateForceX(v2, t):
    if accX == 0:
        calculateAccelerationX(v2,t)
    forceX = mass*accX
    return forceX

def getDistanceEquationX(t, iVelocityX, accX):
    return iVelocityX*t + float(accX)/2*(t*t)

# along y-axis
def calculateAccelerationY(v2, t):
    accY = (v2 - iVelocityY)/t
    return accY

def calculateDistanceY(v2, t):
    if accY == 0:
        calculateAccelerationY(v2,t)
    distanceY = iVelocityY*t + accY*t*t/2
    return distanceY

def calculateForceY(v2, t):
    if accY == 0:
        calculateAccelerationY(v2,t)
    forceY = mass*accY
    return forceY

def getDistanceEquationY(t, iVelocityY, accy):
    return iVelocityY*t + float(accY)/2*(t*t)

# final
def calculateAcceleration():
	return math.sqrt(math.pow(accX,2)+math.pow(accY,2));

def calculateDistance():
	return math.sqrt(math.pow(distanceX,2)+math.pow(distanceY,2));

def calculateForce():
	return math.sqrt(math.pow(forceX,2)+math.pow(forceY,2));

#
accX = 0
accY = 0
acc = 0
distanceX = 0
distanceY = 0
distance = 0
forceX = 0
forceY = 0
force = 0

mass = input("Enter the mass of the object: ")
iVelocityX = input("Enter the initial x-velocity of the object: ")
iVelocityY = input("Enter the initial y-velocity of the object: ")
v2X = input("Enter the current x-velocity of the object: ")
v2Y = input("Enter the current y-velocity of the object: ")
t = input("Enter the duration of motion: ")

accX = calculateAccelerationX(v2X,t)
print("Acceleration-x is "+str(accX)+" m/s2")
accY = calculateAccelerationY(v2Y,t)
print("Acceleration-y is "+str(accY)+" m/s2")
acc = calculateAcceleration()
print("Resultant acceleration is "+str(acc)+" m/s2")

distanceX = calculateDistanceX(v2X,t)
print("X Distance travelled is "+str(distanceX)+" m")
distanceY = calculateDistanceY(v2Y,t)
print("Y Distance travelled is "+str(distanceY)+" m")
distance = calculateDistance()
print("Resultant distance = "+str(distance)+" m")

forceX = calculateForceX(v2X,t)
print("Internal force-x is "+str(forceX)+" N")
forceY = calculateForceY(v2Y,t)
print("Internal force-y is "+str(forceY)+" N")
force = calculateForce()
print("Resultant internal force is "+str(force)+" N")

t1 = nm.arange(0.0,10.0,0.5)
plt.figure(1)
plt.subplot(211)
plt.plot(t1,getDistanceEquationX(t1, iVelocityX, accX),'r--')
plt.legend("Distance aling X axis")

plt.subplot(212)
plt.plot(t1,getDistanceEquationY(t1, iVelocityY, accY),'r--')
plt.legend("Distance aling Y axis")
plt.show()
